src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSize.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Formats.Tar;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+[Trait("Category", "SdkContent")]
+public class ArtifactsSize : SmokeTests
+{
+    private readonly string baselineFilePath = BaselineHelper.GetBaselineFilePath($"ArtifactsSizes/{Config.TargetRid}.txt");
+    private readonly string[] baselineFileContent;
+    private readonly Regex buildVersionPattern = new(@"\b\d+\.\d+\.\d+[-@](alpha|preview|rc|rtm)\.\d(\.\d+\.\d+)?\b");
+
+
+    public ArtifactsSize(ITestOutputHelper outputHelper) : base(outputHelper)
+    {
+        baselineFileContent = File.Exists(baselineFilePath) ? File.ReadAllLines(baselineFilePath) : Array.Empty<string>();
```

Comment:
Thinking about the UX experience in CI I was thinking a missing baseline file should cause an error.  Ideally it would still create the new baseline in this scenario.  Thoughts?

Reply:
I agree with this because it would help alert to a large change.

To clarify, you're suggesting exiting with an error, but create a new baseline, empty baseline file before exiting so that an error would not occur on a following run?

Reply:
Decided to [exit with an error](https://github.com/dotnet/installer/pull/17233/files#diff-ce0a9d7a18f02288a0510a8ec86e3a5f387ad96613e0296f434de93f02c18da7R40) that prompts the user to create the missing file. Once the file has been manually created, the file exists check will pass.